Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -3443,10 +3443,15 @@ RefPtr<CSSValue> ComputedStyleExtractor::valueForPropertyInStyle(const RenderSty
         return createConvertingToCSSValueID(style.inputSecurity());
     case CSSPropertyLeft:
         return positionOffsetValue(style, CSSPropertyLeft, renderer);
-    case CSSPropertyLetterSpacing:
-        if (!style.letterSpacing())
-            return CSSPrimitiveValue::create(CSSValueNormal);
-        return zoomAdjustedPixelValue(style.letterSpacing(), style);
+    case CSSPropertyLetterSpacing: {
+        const Length& spacing = style.computedLetterSpacing();
+        if (spacing.isFixed()) {
+            if (spacing.isZero())
+                return CSSPrimitiveValue::create(CSSValueNormal);
```

Comment:
The computed value is always a length according to the spec: https://drafts.csswg.org/css-text/#letter-spacing-property

I would just remove the special case.

Reply:
The computed value is, but the resolved value isn't.

>  For legacy reasons, a computed letter-spacing of zero yields a resolved value (getComputedStyle() return value) of normal.

Reply:
I got confused by the table: "Computed value: | an absolute length" . Perhaps the spec should refer to the note in the table. Thanks for pointing it out!